packages/http-smoke-testing/src/RouterAdapter/SymfonyRouterAdapter.php

```diff
@@ -27,8 +33,25 @@ public function __construct(RouterInterface $router)
     public function getAllRouteInfo()
     {
         $allRouteInfo = [];
+        $annotationReader = new AnnotationReader();
         foreach ($this->router->getRouteCollection() as $routeName => $route) {
-            $allRouteInfo[] = new RouteInfo($routeName, $route);
+            $annotations = [];
+
+            if ($route->hasDefault('_controller')) {
+                try {
+                    $reflectionMethod = new \ReflectionMethod($route->getDefault('_controller'));
+
+                    foreach ($annotationReader->getMethodAnnotations($reflectionMethod) as $annotation) {
+                        if ($annotation instanceof DataSet || $annotation instanceof Skipped) {
+                            $annotations[] = $annotation;
+                        }
+                    }
+                } catch (\ReflectionException $exception) {
```

Comment:
Thes catch catches only `$reflectionMethod = new \ReflectionMethod($route->getDefault('_controller'));` so it should not be on the `foreach` too.

I would suggest to remove the try/catch block and replace it by checking the existence of givven method as it is the one reason for `\ReflectionMethod` to throw an exception. (https://www.php.net/manual/en/reflectionmethod.construct.php#refsect1-reflectionmethod.construct-errors)

Reply:
I refactored this to use `method_exists()` to avoid controll flow by exception. Thanks for the hint.